src/uu/split/src/split.rs

```diff
@@ -783,6 +807,25 @@ impl Settings {
                 }
             }
         }
+
+        // Make sure that separator is only one UTF8 character (if specified)
+        // defaults to '\n' - newline character
+        // If the same separator (the same value) was used multiple times - `split` should NOT fail
+        // If the separator was used multiple times but with different values (not all values are the same) - `split` should fail
+        let separator = match matches.get_many::<String>(OPT_SEPARATOR) {
+            Some(sep_values) => match sep_values
+                .map(|s| s.to_string())
+                .reduce(|cur, nxt| if cur == nxt { nxt } else { cur + &nxt })
+                .unwrap() // it is safe to just unwrap here since Clap should not return empty ValuesRef<'_,String> in the option from get_many() call
```

Comment:
If you're gonna unwrap, this is all you need I think.
```rust
let first = sep_values.next().unwrap():
if !sep_values.all(|s| s == first) {
    // multiple values given error
}
match first.as_str() {
   ...
}
```
I find the `reduce` a bit hard to follow on second glance, because it is unclear what `cur` represents. It's first a single value and then multiple values glued together, which makes it a bit confusing.

Reply:
no problem, pushed updated version